playground/src/MonacoWrapper.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import * as monacoEditor from 'monaco-editor';
+
+export interface IMonacoWrapperProps {
+  className?: string;
+  style?: React.CSSProperties;
+  value?: string;
+  readOnly?: boolean;
+  language?: string;
+  onChange?: (value: string) => void;
+
+  editorOptions?: monacoEditor.editor.IEditorConstructionOptions;
+}
+
+export interface IMonacoWrapperState {
+  monaco?: typeof monacoEditor;
+  failedToLoad?: boolean;
+}
+
+interface IMonacoWindow extends Window {
+  require: {
+    (paths: string[], callback: (monaco: typeof monacoEditor) => void): void;
+    config: (options: { paths: { [name: string]: string } }) => void
+  };
+  MonacoEnvironment: {
+    getWorkerUrl: (workerId: string, label: string) => void;
+  };
+}
+
+const MONACO_BASE_URL: string = 'https://cdnjs.cloudflare.com/ajax/libs/monaco-editor/0.14.3/min/';
+
+export class MonacoWrapper extends React.Component<IMonacoWrapperProps, IMonacoWrapperState> {
+  private static _initializePromise: Promise<typeof monacoEditor>;
+
+  private _editor: monacoEditor.editor.IStandaloneCodeEditor | undefined;
+  private get _value(): string | undefined {
+    if (this._editor) {
+      return this._editor.getValue();
+    } else {
+      return undefined;
+    }
+  }
+
+  private static _initializeMonaco(): Promise<typeof monacoEditor> {
+    if (!MonacoWrapper._initializePromise) {
+      MonacoWrapper._initializePromise = new Promise(
+        (resolve: (monaco: typeof monacoEditor) => void, reject: (error: Error) => void ) => {
+          const monacoWindow: IMonacoWindow = window as IMonacoWindow;
+          monacoWindow.require.config({ paths: { 'vs': `${MONACO_BASE_URL}vs/` }});
+
+          monacoWindow.MonacoEnvironment = {
+            getWorkerUrl: (workerId, label) => {
+              return `data:text/javascript;charset=utf-8,${encodeURIComponent(
+                'self.MonacoEnvironment = {' +
+                  `baseUrl: '${MONACO_BASE_URL}'` +
+                '};' +
+                `importScripts('${MONACO_BASE_URL}vs/base/worker/workerMain.js');`
+              )}`;
+            }
+          };
+
+          monacoWindow.require(['vs/editor/editor.main'], (monaco) => {
+            if (monaco) {
+              resolve(monaco);
+            } else {
+              reject(new Error('Unable to load Monaco editor'));
+            }
+          });
+        }
+      );
+    }
+
+    return MonacoWrapper._initializePromise;
+  }
+
+  constructor(props: IMonacoWrapperProps) {
+    super(props);
+
+    this.state = {};
+  }
+
+  public componentDidMount(): void {
+    MonacoWrapper._initializeMonaco().then((monaco) => {
+      this.setState({ monaco });
+      window.addEventListener('resize', this._updateLayout.bind(this));
```

Comment:
>window [](start = 6, length = 6)

Should should ideally make sure that `componentWillUnmount()` hasn't executed already; otherwise there's a small race condition here